src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -126,7 +178,14 @@ public static class SymbolDisplay
 
             var builder = ArrayBuilder<SymbolDisplayPart>.GetInstance();
             var visitor = new SymbolDisplayVisitor(builder, format, semanticModelOpt, positionOpt);
-            symbol.Accept(visitor);
+            if (topLevelNullabilityOpt.HasValue)
+            {
+                visitor.VisitWithNullability((TypeSymbol)symbol, topLevelNullabilityOpt.GetValueOrDefault());
```

Comment:
The hard cast seems problematic here: it looks like you were previously allowed to take an ISymbol from VB and pass it into this to format it as a C# symbol?

Reply:
It's also independently bad because in Release builds your debug assert won't fire and I can make you throw here. :smile: